bundles/fr.adaussy.permadeler.model.design.tests/src/fr/adaussy/permadeler/model/design/services/ValidationServiceTest.java

```diff
@@ -0,0 +1,211 @@
+
+		boolean validation = ValidationService.validatePlantationCompatibility(TestServices.area);
+		assertTrue(validation == false);
+	}
+
+	/**
+	 * Test for the service {@link ValidationService#validatePlantationCompatibility(Area)}: validate an area
+	 * which contain 2 plantations with positive affinity link
+	 */
+	@Test
+	public void validatePlantationCompatibilityCompatible() {
+		TestServices.createKnowledgeBase();
+		TestServices.createPlantations();
+		TestServices.plantPlantations(
+				new Plantation[] {TestServices.plantations[0], TestServices.plantations[2] });
+
+		boolean validation = ValidationService.validatePlantationCompatibility(TestServices.area);
+		assertTrue(validation == true);
+	}
+
+	/**
+	 * Test for the service {@link ValidationService#validatePlantationCompatibility(Area)}: validate an area
+	 * which contain 2 plantations with no compatibility link in knowledge base
+	 */
+	@Test
+	public void validatePlantationCompatibilityNoLink() {
+		TestServices.createKnowledgeBase();
+		TestServices.createPlantations();
+		TestServices.plantPlantations(
+				new Plantation[] {TestServices.plantations[1], TestServices.plantations[2] });
+
+		boolean validation = ValidationService.validatePlantationCompatibility(TestServices.area);
+		assertTrue(validation == true);
+	}
+
+	/**
+	 * Test for the service {@link ValidationService#validatePlantationCompatibility(Area)}: validate an area
+	 * which contain 2 plantations with no genuses
+	 */
+	@Test
+	public void validatePlantationCompatibilityNoGenus() {
+		TestServices.createKnowledgeBase();
+		TestServices.plantPlantations(TestServices.plantations);
+
+		boolean validation = ValidationService.validatePlantationCompatibility(TestServices.area);
+		assertTrue(validation == true);
+	}
+
+	/**
+	 * Test for the service {@link ValidationService#validatePlantationCompatibility(Area)}: validate an area
+	 * which contain no plantations
+	 */
+	@Test
+	public void validatePlantationCompatibilityNoPlantation() {
+		TestServices.createKnowledgeBase();
+		TestServices.createPlantations();
+
+		boolean validation = ValidationService.validatePlantationCompatibility(TestServices.area);
+		assertTrue(validation == true);
+	}
+
+	/**
+	 * Test for the service {@link ValidationService#validatePlantationCompatibility(Area)}: case with no
+	 * existing knowledge base
+	 */
+	@Test
+	public void validatePlantationCompatibilityNoCompatibilityMatrix() {
+		TestServices.createPlantations();
+		TestServices.plantPlantations(TestServices.plantations);
+
+		boolean validation = ValidationService.validatePlantationCompatibility(TestServices.area);
+		assertTrue(validation == true);
+	}
+
+	/**
+	 * Variables and methods used in ValidationServiceTest
+	 */
+	private static class TestServices {
+		private final static PermadelerFactory sFactory = PermadelerFactory.eINSTANCE;
+
+		private static final int nbPlantations = 3;
+
+		private static Root root = sFactory.createRoot();
```

Comment:
All attributes should not be static.
You should instantiate in each test a "InputModelHelper inputModel = new InputModelHelper()"

Then either access attribute using get or direct attribute

Reply:
ok, except for nbPlantations and sFactory which are used only in the InputModelHelper class and for which it does make sense to be be static. is that right ?

Reply:
Yes that's right

Reply:
ok